### ISSUE TITLE
1.9.3 broke HomeKit in HASS connecting to go2rtc

### ISSUE THREAD

**Issue by artinbastani** [AlexxIT/go2rtc#1180]
I have a logitech circle 2 camera connected to home assistant as a HomeKit device.

Until 1.9.2 I was add my camera in go2rtc as a "hass:" device. And it worked perfectly

In 1.9.3 this no longer works. It never streams.

When i restored from my prior backup I was able to get this functionality back.

**Comment by petrij98** [comment]
It sounds like I've got a very similar issue. However, I don't believe it's for every camera model. I have 3 different Homekit cameras with 2 working and 1 that's not. Not sure what's causing the issue; I was gonna dig through my logs and see what I could come up with. I didn't manage to make a backup of my 1.9.2 instance, so I feel like I'm stuck until a patch is out or my good ol' docker server's about to take on another app full-time.

**Comment by rouxdoo** [comment]
I just started to transition my homekit cams into HASS and my Circle View isn't streaming in go2rtc. 
[hls] can't get init id=xFH7V6LR is the only log error.

**Comment by AlexxIT** [comment]
I don't know. Maybe this is related to increased default bitrate for HomeKit source. And your camera can't provide it.
You can change bitrate manually, but it works only for `homekit` source, not work for `hass` source.
You can find your `homekit` related source in the WebUI > Add > Home Assistant.

https://github.com/AlexxIT/go2rtc/issues/851

**Comment by rouxdoo** [comment]
I did what you suggested and that fixed it so thank you! That also fixed another problem I was having - I was trying to use hass source for my cameras and simply could not export (pair) to Apple home. Using homekit source made pairing work so you might want to update the documentation in the homekit module section to spell out that passing through homekit camera as homekit source requires using homekit source in the feed config rather than the hass URL. Thanks again!

**Comment by AlexxIT** [comment]
I need to know what maximum bitrate works for you. New default 4096K. Try to find best working maximum.

**Comment by rouxdoo** [comment]
I have a logi circle view cam, a logi circle view doorbell and an Eve Outdoor Cam. The Logitechs bomb out above 1024 (and shows the no-play icon for a few seconds before streaming) but work at that. The Eve seems fine with #bitrate=4096K. I'll be happy to try other values if you want.

**Comment by petrij98** [comment]
This resolved my issue. Thanks @AlexxIT! If you can, I would try to add that as a footnote somewhere in the documentation. It seems like this issue only affects specific homekit cameras, but not all.

**Comment by AlexxIT** [comment]
I think best solution will be to check camera brand from HomeKit protocol. And autoset supported bitrate value.
But I need someone familiar with the go language to log this response from the camera.
Because right now the protocol is not logged in any way.

**Comment by dipseth** [comment]
I am having the same issue even after adjusting the bitrate parameter.  In 1.9.4 I see the camera load briefly in the network map with 0B sent to to homekit, before it  disconnecting.:


![image](https://github.com/AlexxIT/go2rtc/assets/4229974/7ea29175-8832-48b3-894e-dc27b3621ac7)

**Comment by arodey** [comment]
Does a Logi Circle 2 need to be on the special Homekit firmware to work properly?

**Comment by dipseth** [comment]
> Does a Logi Circle 2 need to be on the special Homekit firmware to work properly?

Yes I believe so.

**Comment by arodey** [comment]
Okay that may be my problem.  Is there a work around if I don't have an Apple device?

**Comment by dipseth** [comment]
> Okay that may be my problem.  Is there a work around if I don't have an Apple device?

Yes I believe Home Assistant can act as your Homekit server and you can pair that way.  I'm doing this with 2 different circle cameras.  On second thought I believe I have one pair through HA and through go2rtc (then to frigate)

**Comment by arodey** [comment]
I have them added into Home Assistant with the built in Homekit server.  I can't seem to add them to go2rtc or Frigate though. That's why I was wondering if I need to update the firmware of the cameras to enable Homekit Secure Video. My understanding is once I do this I will lose the ability to use the Logi Circle app and website. I am looking for confirmation before I do something that can't be undone.

**Comment by Vvass-code** [comment]
I root-caused this on a Logi Circle 2 paired directly as a `homekit://` source (not via `hass:`) — full write-up and patch in #2322.

Short version: **three cumulative bugs**, all three needed.

1. `pkg/hap/tlv8/tlv8.go` — the Circle 2 encodes its audio `CodecType` on 2 bytes where the spec says 1; `unmarshalValue` rejects it, so `GetMedias()` returns nil silently.
2. `pkg/homekit/producer.go` — the camera exposes two `00000110` services and the first is a stub with no `VideoAttrs`; `acc.GetCharacter()` returns that one.
3. `pkg/homekit/helpers.go` — the resolution picker compares `Width > || Height >`, so 1024x768 beats 1280x720, and the firmware rejects 4:3.

One thing worth flagging for anyone landing here from a search: **`#bitrate=` did not fix it on my Circle 2.** I tried 299K through 4096K with no effect. It clearly helps the Circle View users above, so it may well be a second, separate issue — but it is not the whole story for this camera, and the thread currently reads as if it were.

Also, on @AlexxIT's earlier point about the protocol not being logged anywhere: `Client.PutCharacters()` reads the HTTP response body and discards it. The camera was replying `207` with `{"aid":1,"iid":22,"status":-70410}` (*invalid value in a write request*) the whole time. Surfacing that one response would make this class of bug self-diagnosing for every camera, not just this one.

Verified on `v1.9.14`: stream works, 1.66 MB of MP4 in 20 s, H264 1280x720.
